src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -286,18 +348,26 @@ private int recordExitStatus(FilePath workspace, int code) throws IOException, I
     private static final class getOsType extends MasterToSlaveCallable<OsType,RuntimeException> {
         @Override public OsType call() throws RuntimeException {
             if (Platform.isDarwin()) {
-              return OsType.DARWIN;
+                return OsType.DARWIN;
             } else if (Platform.current() == Platform.WINDOWS) {
-              return OsType.WINDOWS;
+                return OsType.WINDOWS;
             } else if(Platform.current() == Platform.UNIX && System.getProperty("os.name").equals("z/OS")) {
-              return OsType.ZOS;  
+                return OsType.ZOS;
             } else {
-              return OsType.UNIX; // Default Value
+                return OsType.UNIX; // Default Value
             }
         }
         private static final long serialVersionUID = 1L;
     }
 
+    private static final class getArchitecture extends MasterToSlaveCallable<String,RuntimeException> {
+        @Override public String call() throws RuntimeException {
+            // Note: This will only determine the architecture of the JVM.
```

Comment:
nit: I would update the comment to mention that the value will be either `32` or `64`.
```suggestion
            // Note: This will only determine the architecture of the JVM. The result will be "32" or "64".
```
Maybe worth parsing the property into an enum.